Userland/Libraries/LibGfx/ImageFormats/FLICLoader.cpp

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright (c) 2024, Nicolas Ramz <nicolas.ramz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Memory.h>
+#include <AK/MemoryStream.h>
+#include <LibGfx/ImageFormats/FLICLoader.h>
+
+namespace Gfx {
+
+static constexpr size_t const flic_header_size = 128;
+static constexpr size_t const chunk_header_size = 6;
+
+struct Format {
+    enum : u16 {
+        FLI = 0xAF11,
+        FLC = 0xAF12,
+    };
+};
+
+struct ChunkType {
+    enum : u16 {
+        // sub_chunks
+        COLOR_256 = 0x4,
+        COLOR_64 = 0xB,
+        FLI_COPY = 0x10,
+        DELTA_FLI = 0xC,
+        // main chunks
+        PREFIX_TYPE = 0xF100,
+        SCRIPT_CHUNK = 0xF1E0,
+        FRAME_TYPE = 0xF1FA,
+        SEGMENT_TABLE = 0xF1FB,
+        HUFFMAN_TABLE = 0xF1FC
+    };
+};
+
+struct Chunk {
+    ByteBuffer data;
+    u16 type;
+};
+
+struct FLICFrameDescriptor {
+    Color color_map[256];
+    Vector<NonnullOwnPtr<Chunk>> chunks;
+};
+
+struct FLICLoadingContext {
+    FLICLoadingContext(FixedMemoryStream stream)
+        : stream(move(stream))
+    {
+    }
+
+    enum State {
+        NotDecoded = 0,
+        FrameDescriptorsLoaded,
+        FrameComplete,
+    };
+
+    State state { NotDecoded };
+    enum ErrorState {
+        NoError = 0,
+        FailedToDecodeAllFrames,
+        FailedToLoadFrameDescriptors,
+    };
+    ErrorState error_state { NoError };
+    u16 format;
+    u16 width { 0 };
+    u16 height { 0 };
+    u16 color_depth { 0 };
+    u16 frame_count { 0 };
+    u16 flags { 0 };
+    u32 duration { 0 };
+
+    Vector<NonnullOwnPtr<FLICFrameDescriptor>> frames {};
+    RefPtr<Gfx::Bitmap> frame_buffer;
+    FixedMemoryStream stream;
+};
+
+static ErrorOr<u16> decode_flic_type(FixedMemoryStream& stream)
+{
+    TRY(stream.seek(4, SeekMode::FromCurrentPosition));
```

Comment:
```suggestion
    [[maybe_unused]] auto const file_size = TRY(stream.read_value<LittleEndian<u32>>());
```
This makes it clearer why you're discarding bytes, and also remove the need for a seekeable stream.